jetty-client/src/main/java/org/eclipse/jetty/client/http/HttpReceiverOverHTTP.java

```diff
@@ -235,6 +235,10 @@ private boolean parse()
 
             if (complete)
             {
+                HttpExchange httpExchange = getHttpExchange();
+                if (httpExchange != null && httpExchange.getResponse().getStatus() == HttpStatus.SWITCHING_PROTOCOLS_101)
```

Comment:
Previously the test for upgrade was: `connection != endPoint.getConnection()`
Now it is just 101 response status.   Don't we support upgrade like paths (eg for http2) that are not 101s?

Reply:
@gregw but this class is the HTTP/1.1 specific implementation so it cannot be anything else than 101.
The reason to not do the check to see if the connection was upgraded already is because the 101 response may be finished before the request, so the upgrade did not happen yet (but we know it will since it's a 101).